tests/rptest/tests/datalake/schema_evolution_test.py

```diff
@@ -0,0 +1,216 @@
+        }
+    ]
+}
+"""
+
+legal_evo_schema_avro = """
+{
+    "type": "record",
+    "name": "VerifierRecord",
+    "fields": [
+        {
+            "name": "verifier_string",
+            "type": "string"
+        },
+        {
+            "name": "ordinal",
+            "type": "long"
+        },
+        {
+            "name": "another",
+            "type": "float"
+        }
+    ]
+}
+"""
+
+illegal_evo_schema_avro = """
+{
+    "type": "record",
+    "name": "VerifierRecord",
+    "fields": [
+        {
+            "name": "verifier_string",
+            "type": "double"
+        },
+        {
+            "name": "ordinal",
+            "type": "long"
+        }
+    ]
+}
+"""
+
+
+class SchemaEvolutionTest(RedpandaTest):
+    def __init__(self, test_context):
+        super(SchemaEvolutionTest, self).__init__(
+            test_context=test_context,
+            num_brokers=1,
+            si_settings=SISettings(test_context,
+                                   cloud_storage_enable_remote_read=False,
+                                   cloud_storage_enable_remote_write=False),
+            extra_rp_conf={
+                "iceberg_enabled": True,
+                "iceberg_catalog_commit_interval_ms": 5000,
+            },
+            schema_registry_config=SchemaRegistryConfig())
+
+    def avro_stream_config(self,
+                           topic,
+                           subject,
+                           mapping: str = f"""
+                    root.verifier_string = uuid_v4()
+                    root.ordinal = counter()
+                    """):
+        return {
+            "input": {
+                "generate": {
+                    "mapping": mapping,
+                    "interval": "",
+                    "count": 1000,
+                    "batch_size": 1
+                }
+            },
+            "pipeline": {
+                "processors": [{
+                    "schema_registry_encode": {
+                        "url": self.redpanda.schema_reg().split(",")[0],
+                        "subject": subject,
+                        "refresh_period": "10s"
+                    }
+                }]
+            },
+            "output": {
+                "redpanda": {
+                    "seed_brokers": self.redpanda.brokers_list(),
+                    "topic": topic,
+                }
+            }
+        }
+
+    def setUp(self):
+        pass
+
+    def _create_schema(self, subject: str, schema: str, schema_type="avro"):
+        rpk = RpkTool(self.redpanda)
+        with tempfile.NamedTemporaryFile(suffix=f".{schema_type}") as tf:
+            tf.write(bytes(schema, 'UTF-8'))
+            tf.flush()
+            rpk.create_schema(subject, tf.name)
+
+    @cluster(num_nodes=4, log_allow_list=["Schema \d+ already exists"])
+    @matrix(
+        cloud_storage_type=supported_storage_types(),
+        query_engine=[
+            QueryEngineType.SPARK,
+            # QueryEngineType.TRINO,
```

Comment:
?

Reply:
sigh. added a remove column test case to dull the pain.